### PR TITLE
Closes #82 vignette update

### LINKED ISSUE #82: Documentation:  Update the vignette
### Please select a category the issue is focused on?

User Guides

### Let us know where something needs a refresh or put your idea here!

Update the vignette to implement David's latest [comments](https://phuseaccount.sharepoint.com/:w:/r/sites/Pharmaverse-admiral/_layouts/15/Doc.aspx?sourcedoc=%7B33A54EDA-C7FF-45D9-8553-CE694CE8184D%7D&file=20240728_revised_admiralpeds_vignette.docx&action=default&mobileredirect=true)

### PR DESCRIPTION
**Pull request by Fanny-Gautier**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiralpeds (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiralpeds site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by Fanny-Gautier** [comment]
@zdz2101 , @manciniedoardo ,
I have implemented most of David's [comments](https://phuseaccount.sharepoint.com/:w:/r/sites/Pharmaverse-admiral/_layouts/15/Doc.aspx?sourcedoc=%7B33A54EDA-C7FF-45D9-8553-CE694CE8184D%7D&file=20240728_revised_admiralpeds_vignette.docx&action=default&mobileredirect=true). Not being an English native speaker and having overwritten most of the advs vignette, could you please carefully review it ensuring that no important information has been dropped ?
Please let me know if further comments need to be implemented.
Many thanks.

File: vignettes/advs.Rmd [hunk 1]
```diff
@@ -309,26 +302,23 @@ advs <- advs %>%
...
-functions we took the simple approach to use the nearest age from the
-metadata, rather than apply some form of interpolation. An alternative
-method you could use would be to derive current age according to the
-exact same intervals as the metadata in order for an exact join, but in
-our examples we just kept this by days.
+Note that some growth reference files we use here have different units and intervals. 
+For example when you look at the "by Age" charts, WHO provides this by day,
+but the CDC gives this every month (25.5, 26.5, etc.). So, converting all ages 
+to days leaves the challenge of joining when the patient's current age falls 
+between the monthly CDC reference data. In our package functions, we used the
```
[the remark this answers is not in the document]
Comment (zdz2101) [reply vignettes/advs.Rmd:317]: I'm not sure what's the best way to reword exactly what we did or just get rid of this paragraph in its entirety

**Review by zdz2101 — COMMENTED** [review]
But for the most part I like these quality of life changes made, it reads smoother

File: vignettes/advs.Rmd [hunk 2]
```diff
@@ -168,33 +167,27 @@ dataset_vignette(
...
-The above example only shows the parameter BMI, but similar follows for the
-other parameters like weight and height (except no Sigma and P95 for those),
-and for height we are able to combine WHO and CDC metadata as there is no WHO
-adjustment needed.
-Note that the head circumference parameter metadata comes only from
-the WHO standards, as no equivalent for \>2 years of age are provided by CDC.
+The above example only shows the parameter BMI, but this follows other parameters
+like weight and height. We can combine WHO and CDC metadata for height as a
+WHO adjustment is unnecessary.
+Note that the head circumference parameter metadata comes only from the WHO standards,
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:176]:
```suggestion
Note that the head circumference parameter metadata comes only from the WHO reference data,
```

Comment (rossfarrugia) [reply vignettes/advs.Rmd:176]: Given David's other comments/updates

File: vignettes/advs.Rmd [hunk 3]
```diff
@@ -168,33 +167,27 @@ dataset_vignette(
...
+as CDC provides no equivalent for children \> 2 years of age.
 
 ### Reference Files for Weight by Length/Height
 
-WHO provides additional standards for weight by body length (recumbent length),
-instead of age. For these we again combined the metadata for sex so that we have
-a single reference file. 
+WHO provides additional standards for weight-for-length (recumbent length),
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:181]:
```suggestion
WHO provides additional reference data for weight-for-length (recumbent length),
```

File: vignettes/advs.Rmd [hunk 4]
```diff
@@ -424,31 +414,33 @@ advs <- advs %>%
...
-For BMI parameters only, you'll notice from the code above that we have
-set `bmi_cdc_correction = TRUE`. This is because we opted to use the CDC
-developed extended percentiles (\>95%) to monitor high BMI values here.
-If you left this argument as default `FALSE` value then from the CDC metadata
-only the 2000 CDC Growth Chart for BMI would be used.
-
-For any weight-based indicator (including BMI), you'll also see in the code
+Once again, it is essential to remember the importance of the metadata being
+supplied to the above function via the `meta_criteria` argument.
+Our default case from this package uses WHO standards for children \< 2 years of
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:422]:
```suggestion
Our default case from this package uses WHO reference data for children \< 2 years of
```

File: vignettes/advs.Rmd [hunk 5]
```diff
@@ -168,33 +167,27 @@ dataset_vignette(
...
+Note that the head circumference parameter metadata comes only from the WHO standards,
+as CDC provides no equivalent for children \> 2 years of age.
 
 ### Reference Files for Weight by Length/Height
 
-WHO provides additional standards for weight by body length (recumbent length),
-instead of age. For these we again combined the metadata for sex so that we have
-a single reference file. 
+WHO provides additional standards for weight-for-length (recumbent length), 
+instead of age. Again, we combined the metadata for sex to create a single
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:182]:
```suggestion
instead of age. Again, we combined the metadata for sex to create a single
```
remove any blank spaces at the end of lines

File: vignettes/advs.Rmd [hunk 6]
```diff
@@ -168,33 +167,27 @@ dataset_vignette(
...
-children are able to stand then BMI by age is a more appropriate measure to use.
-However, in our template we do give the metadata values variable a more generic
-name as `HEIGHT_LENGTH` so that any user could equally choose to pass weight by
-height files if they so preferred.
+There are also weight-by-height files available, but upon advice from a retired 
+CDC expert in this area, we chose only to include the weight-for-length files, 
+as once children can stand, BMI by age is a more appropriate measure to use.
+However, in our template, we give the metadata values variable a generic name,
+`HEIGHT_LENGTH`, so that any user could equally choose to pass weight-for-height
+files if preferred.
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:190]:
```suggestion
files if preferred.
```

File: vignettes/advs.Rmd [hunk 7]
```diff
@@ -221,8 +214,8 @@ dataset_vignette(
 
 ## Initial ADVS Set-up {#advs_start}
 
-The next step would be reading in the source data, merging ADSL
-variables and deriving the usual ADVS analysis variables. The
+The following steps are to read in the source data, merge ADSL variables,
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:217]:
```suggestion
The following steps are to read in the source data, merge ADSL variables,
```

File: vignettes/advs.Rmd [hunk 8]
```diff
@@ -221,8 +214,8 @@ dataset_vignette(
 
 ## Initial ADVS Set-up {#advs_start}
 
-The next step would be reading in the source data, merging ADSL
-variables and deriving the usual ADVS analysis variables. The
+The following steps are to read in the source data, merge ADSL variables, 
+and derive the usual ADVS analysis variable. The
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:218]:
```suggestion
and derive the usual ADVS analysis variables. The
```

File: vignettes/advs.Rmd (hunk 1 above)
[the remark this answers is not in the document]
Comment (rossfarrugia) [reply vignettes/advs.Rmd:317]: i would remove all this section (lines 313-321). i think its a carry over from before we added the interpolation piece above which now covers this

File: vignettes/advs.Rmd [hunk 9]
```diff
@@ -375,14 +365,14 @@ advs <- advs %>%
...
 ## Derive Additional Parameters for Anthropometric indicators {#derive_params}
 
-Now we get to the most important section where we show how to create the
-new records for each derived pediatrics parameter. Only certain examples
-are included here, but you'll find more parameters in the example
-template script referenced at the bottom of this vignette.
+Now, we get to the most important section, which shows how to create new records
+for each derived pediatrics parameter.
+Only specific examples are included here, but you'll find more parameters in the
+example template script referenced at the bottom of this vignette.
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:371]:
```suggestion
example template script referenced at the bottom of this vignette.
```

File: vignettes/advs.Rmd [hunk 10]
```diff
@@ -424,31 +414,33 @@ advs <- advs %>%
...
-developed extended percentiles (\>95%) to monitor high BMI values here.
-If you left this argument as default `FALSE` value then from the CDC metadata
-only the 2000 CDC Growth Chart for BMI would be used.
-
-For any weight-based indicator (including BMI), you'll also see in the code
+Once again, it is essential to remember the importance of the metadata being
+supplied to the above function via the `meta_criteria` argument.
+Our default case from this package uses WHO standards for children \< 2 years of
+age and CDC reference data for children \>= 2 years of age. As the user, you can 
+use other (e.g., IOTF) reference data files created with the LMS method.
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:424]:
```suggestion
use other (e.g., International Obesity Task Force) reference data files created with the LMS method.
```

Comment (rossfarrugia) [reply vignettes/advs.Rmd:424]: can also then remove this from wordlist file

File: vignettes/advs.Rmd [hunk 11]
```diff
@@ -462,18 +454,18 @@ dataset_vignette(
...
 This is explained thoroughly in the next section.
 
 ### Derived Weight by Length/Height
 
-Parameters for Weight by Length/Height z-score and percentile for children under
-2 years of age could be achieved using the following code, where it assumes the
-"HEIGHT" parameter is always collected as body length (recumbent length).
+The following code is for Weight by Length/Height z-score and percentile for
+children \< 2 years of age, which assumes the "HEIGHT" parameter is always
+collected as recumbent length.
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:465]:
```suggestion
collected as recumbent length.
```

File: vignettes/advs.Rmd [hunk 12]
```diff
@@ -501,24 +493,22 @@ advs <- advs %>%
   )
 ```
 
-We consulted with an acting pediatrician who advised that rarely investigators
-are guided as to which measure to use. It depends mainly on the age
-at which the child is able to stand (2 years on average) as well
-as the preference of the measurer.
+We consulted with a pediatrician with extensive experience working with length 
+and height data, who advised that analysts rarely know whether standing height 
+or recumbent length was measured. It depends mainly on the age at which the
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:498]:
```suggestion
or recumbent length was measured. It depends mainly on the age at which the
```

File: vignettes/advs.Rmd [hunk 13]
```diff
@@ -501,24 +493,22 @@ advs <- advs %>%
   )
 ```
 
-We consulted with an acting pediatrician who advised that rarely investigators
-are guided as to which measure to use. It depends mainly on the age
-at which the child is able to stand (2 years on average) as well
-as the preference of the measurer.
+We consulted with a pediatrician with extensive experience working with length
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:496]:
```suggestion
We consulted with a pediatrician with extensive experience working with length
```

File: vignettes/advs.Rmd [hunk 14]
```diff
@@ -501,24 +493,22 @@ advs <- advs %>%
   )
 ```
 
-We consulted with an acting pediatrician who advised that rarely investigators
-are guided as to which measure to use. It depends mainly on the age
-at which the child is able to stand (2 years on average) as well
-as the preference of the measurer.
+We consulted with a pediatrician with extensive experience working with length 
+and height data, who advised that analysts rarely know whether standing height
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:497]:
```suggestion
and height data, who advised that analysts rarely know whether standing height
```

**Review by rossfarrugia — CHANGES_REQUESTED** [review]
@Fanny-Gautier some minor changes spotted. I'll try rendering the vignette also now and see if i spot anything more but i don't see anything contraversial in the updates David suggested. All seem fine to me.

File: vignettes/advs.Rmd [hunk 15]
```diff
@@ -79,28 +73,33 @@ metadata for this package are as follows:
...
-
-Additionally, as the CDC growth charts only offer monthly intervals we
-took advice from an ex-CDC expert in this area and made the decision in
-our template to add interpolation code so that we have a record for each
-day of age. When converting the CDC months into days we use rounding
-to bring them to a whole number of days. If you would prefer to rely
-solely on the monthly chart here, then you could remove the rounding
-and the call to the `derive_interp_records()` function.
+As growth rates and patterns by age and sex, reference values differ by these
+two characteristics.
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:77]: @Fanny-Gautier @zdz2101 does this sentence read well to you? it doesn't make sense to me. i found the old wording clearer but might just be me

File: vignettes/advs.Rmd [hunk 16]
```diff
@@ -79,28 +73,33 @@ metadata for this package are as follows:
...
+two characteristics. 
+
+For the various measures, we use the WHO reference data for children <2 years 
+of age (\<730.5 days) and CDC growth charts for children >=2 years of age 
+(\>=730.5 days).
+This is in accord with the [CDC
+recommendation](https://www.cdc.gov/growthcharts/who_charts.htm). The reference 
+data for both charts include the LMS parameters, in which L is the Box-Cox 
+transformation for normality, M is the median, and S is the coefficient of 
+variation (S).
```
Comment (rossfarrugia) [inline vignettes/advs.Rmd:86]:
```suggestion
variation. 
```
doesn't need the brackets

File: vignettes/advs.Rmd (hunk 15 above)
Comment (Fanny-Gautier) [reply vignettes/advs.Rmd:77]: I am not a native English speaker, I had first to translate it to ensure I properly understood it... We can revert it or change it again if you prefer. I am totally fine with this option ;-)

Comment (rossfarrugia) [reply vignettes/advs.Rmd:77]: Revert it please. I think its missing a word or 2 for it to read properly.

Comment (Fanny-Gautier) [reply vignettes/advs.Rmd:77]: I think that the correct sentence should have been "As growth rates and patterns **differ** by age and sex, reference values differ by these two characteristics. "

What do you think instead of "As growth rates and patterns differ between sex, different reference values are used accordingly and for different age groups.", as the part "and for different age groups." is also a bit weird.

**Review by rossfarrugia — APPROVED** [review]
